lib/maths/time_series/CTimeSeriesModel.cc

```diff
@@ -1007,11 +1046,31 @@ bool CUnivariateTimeSeriesModel::uncorrelatedProbability(
 
         double pAnomaly;
         std::tie(pOverall, pAnomaly) = m_AnomalyModel->probability(pSingleBucket, pOverall);
+        if (pAnomaly < common::LARGEST_SIGNIFICANT_PROBABILITY) {
+            result.s_AnomalyScoreExplanation.s_AnomalyType =
+                (m_AnomalyModel->sumPredictionError() > (0.0))
+                    ? common::SAnomalyScoreExplanation::E_SPIKE
+                    : common::SAnomalyScoreExplanation::E_DIP;
+            result.s_AnomalyScoreExplanation.s_AnomalyLength =
+                std::min(m_AnomalyModel->length(time), 12UL);
```

Comment:
I'm not sure this should be min. I would make this something like:
```
result.s_AnomalyScoreExplanation.s_MultiBucketImpact != 0 ? std::max(m_AnomalyModel->length(time), 12UL) : m_AnomalyModel->length(time);
```
i.e. if we know multi-bucket is important we know last 12 buckets are relevant, but it could be longer. Also, 12 feels like a magic number. I would add an accessor to `m_MultibucketFeature` which returns `m_SlidingWindow.capacity()`.